src/bdd_params/impl_bdd_parameter_encoder.rs

```diff
@@ -8,6 +8,12 @@ impl BddParameterEncoder {
     /// Create a new `BddParameterEncoder` based on information given in a `BooleanNetwork`.
     pub fn new(bn: &BooleanNetwork) -> BddParameterEncoder {
         let mut bdd = BddVariableSetBuilder::new();
+        return new_with_custom_variables(bn, bdd);
+    }
+
+    /// Create a new `BddParameterEncoder` based on information given in a `BooleanNetwork`
+    /// and with variables already provided via `BddVariableSetBuilder`.
+    pub fn new_with_custom_variables(bn: &BooleanNetwork, mut bdd: &BddVariableSetBuilder) -> BddParameterEncoder {
```

Comment:
We cannot use `mut bdd: &BddVariableSetBuilder`:
1. `mut &` is not the same as `&mut`. First means the reference itself can be changed (i.e. I can set it to a different `BddVariableSetBuilder`), the second means that the contents of the referenced object can be changed (i.e. I can add a new variable to the the builder). However...
2. The builder is *consumed* when `bdd.build()` is called at the end of the method. That means `new_with_custom_variables` actually needs to have ownership of the builder, not just a reference. So the correct type should be just `mut bdd: BddVariableSetBuilder` :)